### ISSUE TITLE
~6 bugs in one short video: Modules, Registers showing bogus output, cannot edit watch expressions, watch expression view rule not respected, assembly view not showing interactable registers, watch address location not correct

### ISSUE THREAD

**Issue by seanptmaher** [EpicGamesExt/raddebugger#337]
Debugger version 9.12 ALPHA - Oct 14 2024 [cc2848f] (most recent release on github)

I am debugging a component debug build of Chromium. I can provide you with more info if you'd like.

As the title (which may get clipped) spells out:
~6 bugs in one short video:
- Modules, where it shows some `__elems_` object which is... zero sized types? but which have values?
- Registers showing bogus output, showing various random objects that Idk what they are.
- cannot edit watch expressions, screen flashes red, doesn't give any reason why (it's just on the stack?? I can't edit the memory either because...)
- watch expression view rule not respected, it shows it in hex with the enum name.
- assembly view not showing interactable registers, only those AFTER $rip
- watch address location not correct, as shown in my jumping to that address and it containing garbage.

Here is the video in question:

When there is a "bug," I tend to circle my mouse around it :P 

https://drive.google.com/file/d/1bcjgRgQ1t_0uvXYzWZEL501NHgCGE3fV/

**Comment by ryanfleury** [comment]
Without testing Chromium locally, I believe I've fixed the registers/modules view problems as of 1865c337041d296a2174f92e4ebcb96e8795d2e8.

> watch expression view rule not respected, it shows it in hex with the enum name.
This should be fixed as of 53c624a27cee0416f35c4dc44f2044541500f7ea.

As for the others:

> screen flashes red, doesn't give any reason why
The reason is in the bottom right-hand corner of the window, but it likely just says that committing failed. I'll try to recreate that one locally, I have a suspicion of what is causing that.

> assembly view not showing interactable registers, only those AFTER $rip
I think this is likely just an assembly syntax lexing problem (it looks like the lexer got confused at some point) - to get around it temporarily, you can select the register name and hover it.

> watch address location not correct, as shown in my jumping to that address and it containing garbage.
This is not a bug, as the watch pin "address location" is referring to the spot in disassembly where you've attached that pin, not the evaluated address of the expression. (You can see how that address changes if you drag/drop that pin around). When you navigate to the address of the pin, that address is the location within the disassembly that you've attached that pin to. So when you navigate there and it shows the various byte values, those are the byte values of the disassembly, not of the expression you're evaluating.

**Comment by ryanfleury** [comment]
Okay, that watch window mutation bug should also be fixed as of b3d6b2913e925c74701700a10b3b855ada2da0cb. You should be able to write to that value.

You can't modify in the memory view yet, but this is just a known limitation of the memory view; that UI will be more fleshed out a bit later on.

**Comment by seanptmaher** [comment]
Thanks for the super fast response!

**Comment by seanptmaher** [comment]
In the interest of not creating more issues:

Here's a new video showing stepping being broken when it has to actually step over a function call:

https://drive.google.com/file/d/1EGY2o3tFKJRtaSW8eUXDgVOgz37O2Hbn

Basically, it freezes, "running" the target, but it's actually paused at the correct 'ret' instr.

Then, pausing, and stepping once more will return to the right location.

**Comment by seanptmaher** [comment]
Thanks for your work on this by the way, you're doing god's work.....

**Comment by seanptmaher** [comment]
If you'd rather I create new issues for these minute little things for easier bugtracking, let me know and I'll do that.

Hovering over a stack frame with extensive template use, the types of the parameters cause the name to be super long, pushing the entire signature out of screen. Can that be changed so it's left-justified and won't go wider than the width of my screen, either wrapping or pushing off to the right?

![image](https://github.com/user-attachments/assets/590f231e-7a56-4018-b2f2-9b174e8ac34f)

**Comment by mistymntncop** [comment]
> In the interest of not creating more issues:
> 
> Here's a new video showing stepping being broken when it has to actually step over a function call:
> 
> https://drive.google.com/file/d/1EGY2o3tFKJRtaSW8eUXDgVOgz37O2Hbn
> 
> Basically, it freezes, "running" the target, but it's actually paused at the correct 'ret' instr.
> 
> Then, pausing, and stepping once more will return to the right location.

Might be related to this line stepping bug ?
https://github.com/EpicGamesExt/raddebugger/issues/322

**Comment by ryanfleury** [comment]
Yeah, I would prefer new issues for each separate category of bug - that helps me keep track of things a bit better.

The disassembly lexing bug has been fixed as of e9666e2eddee1665169b6c74f0e9f6660e746f86. I've also fixed the thread callstack tooltips when extremely long autogenerated symbol names are used; there is still no path for trying to make those names more helpful, but I've just stopped the tooltip from overtaking too much. I've fixed #322, so I'll look into that stepping bug now.
